nav2_behavior_tree/plugins/action/remove_passed_goals_action.cpp

```diff
@@ -31,12 +31,44 @@ RemovePassedGoals::RemovePassedGoals(
   viapoint_achieved_radius_(0.5)
 {
   getInput("radius", viapoint_achieved_radius_);
-
-  getInput("global_frame", global_frame_);
-  getInput("robot_base_frame", robot_base_frame_);
+  
   tf_ = config().blackboard->get<std::shared_ptr<tf2_ros::Buffer>>("tf_buffer");
   auto node = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
   node->get_parameter("transform_tolerance", transform_tolerance_);
+
+  if(!getInput("global_frame", global_frame_)) {
+    RCLCPP_INFO(node->get_logger(), "Parameter 'global_frame' not provided by behavior tree xml file, trying to get it from ros2 parameter file");
+    if(!node->get_parameter("global_frame", global_frame_)){
```

Comment:
If you declare it with the default as `node->get_parameter("global_frame").as_string()`, it will never not have a response to `get_parameter` with the default of `node->get_parameter("global_frame")` so this particular log/check is not necessary

Reply:
Fixed.